lib/ukargparse/argparse.c

```diff
@@ -139,3 +140,56 @@ int uk_argnparse(char *argb, __sz maxlen, char *argv[], int maxcount)
 out:
 	return argc;
 }
+
+__sz uk_nextarg_r(const char **argptr, int separator)
+{
+	const char *nsep;
+	const char *arg;
```

Comment:
Would it make sense to add an assert for `separator` checking if it is out of `char` bounds?

Reply:
Hum, ... since it is just forwarded to `strchr`, I'd offload that check to that function.